mlflow/utils/requirements_utils.py

```diff
@@ -426,7 +434,17 @@ def _infer_requirements(model_uri, flavor):
             _PYPI_PACKAGE_INDEX.date,
             unrecognized_packages,
         )
-    return sorted(map(_get_pinned_requirement, packages))
+
+    pinned_requirements = []
+    for package in packages:
+        # special casing for mlflow[gateway] to ensure proper formatting of the extra
+        if package == "mlflow[gateway]":
+            pinned_requirement = _get_pinned_requirement("mlflow", extras=["gateway"])
+        else:
+            pinned_requirement = _get_pinned_requirement(package)
```

Comment:
We can update this to one line though
```suggestion
        pinned_requirement = _get_pinned_requirement("mlflow", extras=["gateway"]) if package == "mlflow[gateway]" else _get_pinned_requirement(package)
```